demo-projects/blog/app/pages/post/[slug].js

```diff
@@ -118,81 +108,100 @@ const Comments = ({ data }) => (
   </div>
 );
 
-const AddComments = ({ users, post }) => {
-  let user = users.filter(u => u.email == 'user@keystonejs.com')[0];
+const AddComments = ({ post }) => {
   let [comment, setComment] = useState('');
 
-  const [createComment] = useMutation(ADD_COMMENT, {
-    update: (cache, { data: data }) => {
-      const { allComments, allUsers, allPosts } = cache.readQuery({
-        query: ALL_QUERIES,
-        variables: { slug: post.slug },
-      });
+  const { data, loading: userLoading, error: userError } = useQuery(gql`
+    query {
+      authenticatedUser {
+        id
+      }
+    }
+  `);
 
-      cache.writeQuery({
-        query: ALL_QUERIES,
-        variables: { slug: post.slug },
-        data: {
-          allPosts,
-          allUsers,
-          allComments: allComments.concat([data.createComment]),
-        },
-      });
-    },
+  const [createComment, { loading: savingComment, error: saveError }] = useMutation(ADD_COMMENT, {
+    refetchQueries: ['AllQueries'],
```

Comment:
Instead of doing the `update` / `writeQuery` dance, I've simplified the demo to just refetch the data after the mutation.